### PR TITLE
Add SSI denylist and tests

### PR DESCRIPTION
**Pull request by andrewlock**
## Summary of changes

Adds the SSI denylist and initial testing

## Reason for change

We want to bail out early if we don't support the platform.

## Implementation details

Add a requirements.json file that specifies supported hosts and some of the patterns we expect to block or allow injection on

## Test coverage

This includes a `requirements_json_test` stage. This will be ported to the one-pipeline once the `auto_inject` code is merged, this is a PoC that proves it works atm

## Other details

Will backport this to 2.x as well

### CODE REVIEW

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (5928)](https://github.com/DataDog/dd-trace-dotnet/tree/8e7e595938ea5a9f610fbf9d1ead066782d80059)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).

**Comment by andrewlock** [comment]
## Benchmarks Report for tracer :snail:

Benchmarks for #5928 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab): 
* 1 benchmarks are **faster**, with geometric mean 1.119
* 2 benchmarks are **slower**, with geometric mean 1.139
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.ActivityBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`StartStopWithChild`|net6.0|7.79μs|40.5ns|199ns|0.0153| 0.00764|0|5.42&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`StartStopWithChild`|netcoreapp3.1|9.78μs|53.2ns|315ns|0.0203| 0.0101|0|5.61&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`StartStopWithChild`|net472|16μs|57.7ns|223ns|1.01| 0.292|0.0948|6.07&nbsp;KB|
|#5928|`StartStopWithChild`|net6.0|7.68μs|42ns|241ns|0.0162| 0.0081|0|5.42&nbsp;KB|
|#5928|`StartStopWithChild`|netcoreapp3.1|9.53μs|52ns|307ns|0.0188| 0.0094|0|5.62&nbsp;KB|
|#5928|`StartStopWithChild`|net472|16.1μs|53.1ns|206ns|1.02| 0.301|0.0978|6.07&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`WriteAndFlushEnrichedTraces`|net6.0|462μs|231ns|866ns|0| 0|0|2.7&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|651μs|228ns|882ns|0| 0|0|2.7&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`WriteAndFlushEnrichedTraces`|net472|836μs|390ns|1.51μs|0.417| 0|0|3.3&nbsp;KB|
|#5928|`WriteAndFlushEnrichedTraces`|net6.0|483μs|212ns|820ns|0| 0|0|2.7&nbsp;KB|
|#5928|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|631μs|561ns|2.17μs|0| 0|0|2.7&nbsp;KB|
|#5928|`WriteAndFlushEnrichedTraces`|net472|848μs|698ns|2.7μs|0.425| 0|0|3.3&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`SendRequest`|net6.0|193μs|1.07μs|6.68μs|0.182| 0|0|18.45&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`SendRequest`|netcoreapp3.1|213μs|1.18μs|9.12μs|0.22| 0|0|20.61&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`SendRequest`|net472|0.00227ns|0.000957ns|0.00371ns|0| 0|0|0&nbsp;b|
|#5928|`SendRequest`|net6.0|187μs|1.03μs|6.53μs|0.193| 0|0|18.45&nbsp;KB|
|#5928|`SendRequest`|netcoreapp3.1|206μs|786ns|3.43μs|0.222| 0|0|20.61&nbsp;KB|
|#5928|`SendRequest`|net472|0.000116ns|0.000116ns|0.000403ns|0| 0|0|0&nbsp;b|
</details>


<details><summary><strong>Benchmarks.Trace.CIVisibilityProtocolWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`WriteAndFlushEnrichedTraces`|net6.0|566μs|3μs|15μs|0.548| 0|0|41.71&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|676μs|2.01μs|7.52μs|0.334| 0|0|41.91&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`WriteAndFlushEnrichedTraces`|net472|860μs|1.87μs|6.73μs|8.45| 2.53|0.422|53.28&nbsp;KB|
|#5928|`WriteAndFlushEnrichedTraces`|net6.0|555μs|2.02μs|7.27μs|0.548| 0|0|41.65&nbsp;KB|
|#5928|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|675μs|3.58μs|17.9μs|0.331| 0|0|41.71&nbsp;KB|
|#5928|`WriteAndFlushEnrichedTraces`|net472|875μs|2.88μs|11.2μs|8.3| 2.62|0.437|53.33&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`ExecuteNonQuery`|net6.0|1.3μs|2.11ns|8.17ns|0.0143| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`ExecuteNonQuery`|netcoreapp3.1|1.71μs|1.78ns|6.66ns|0.0136| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`ExecuteNonQuery`|net472|1.96μs|2.16ns|8.07ns|0.157| 0|0|987&nbsp;B|
|#5928|`ExecuteNonQuery`|net6.0|1.23μs|0.986ns|3.69ns|0.0141| 0|0|1.02&nbsp;KB|
|#5928|`ExecuteNonQuery`|netcoreapp3.1|1.71μs|1.04ns|4.03ns|0.0137| 0|0|1.02&nbsp;KB|
|#5928|`ExecuteNonQuery`|net472|2.03μs|2.57ns|9.96ns|0.156| 0|0|987&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`CallElasticsearch`|net6.0|1.17μs|2.3ns|8.92ns|0.0134| 0|0|976&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`CallElasticsearch`|netcoreapp3.1|1.61μs|1.77ns|6.86ns|0.0136| 0|0|976&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`CallElasticsearch`|net472|2.35μs|2.23ns|8.63ns|0.157| 0.00118|0|995&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`CallElasticsearchAsync`|net6.0|1.29μs|0.852ns|3.19ns|0.0135| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`CallElasticsearchAsync`|netcoreapp3.1|1.66μs|2.31ns|8.95ns|0.0142| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`CallElasticsearchAsync`|net472|2.64μs|2.21ns|8.55ns|0.167| 0|0|1.05&nbsp;KB|
|#5928|`CallElasticsearch`|net6.0|1.18μs|0.77ns|2.98ns|0.0136| 0|0|976&nbsp;B|
|#5928|`CallElasticsearch`|netcoreapp3.1|1.63μs|0.598ns|2.24ns|0.013| 0|0|976&nbsp;B|
|#5928|`CallElasticsearch`|net472|2.55μs|1.45ns|5.63ns|0.157| 0|0|995&nbsp;B|
|#5928|`CallElasticsearchAsync`|net6.0|1.26μs|1.02ns|3.82ns|0.0134| 0|0|952&nbsp;B|
|#5928|`CallElasticsearchAsync`|netcoreapp3.1|1.58μs|1.69ns|6.33ns|0.0135| 0|0|1.02&nbsp;KB|
|#5928|`CallElasticsearchAsync`|net472|2.6μs|3.14ns|12.1ns|0.166| 0.00129|0|1.05&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`ExecuteAsync`|net6.0|1.24μs|0.938ns|3.63ns|0.0131| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`ExecuteAsync`|netcoreapp3.1|1.66μs|0.939ns|3.52ns|0.0124| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`ExecuteAsync`|net472|1.78μs|1.08ns|4.17ns|0.145| 0|0|915&nbsp;B|
|#5928|`ExecuteAsync`|net6.0|1.23μs|2.29ns|8.88ns|0.0134| 0|0|952&nbsp;B|
|#5928|`ExecuteAsync`|netcoreapp3.1|1.59μs|1.22ns|4.56ns|0.0127| 0|0|952&nbsp;B|
|#5928|`ExecuteAsync`|net472|1.8μs|1.73ns|6.68ns|0.145| 0|0|915&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`SendAsync`|net6.0|4.06μs|1.25ns|4.66ns|0.0304| 0|0|2.22&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`SendAsync`|netcoreapp3.1|5.12μs|1.7ns|6.6ns|0.0352| 0|0|2.76&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`SendAsync`|net472|7.77μs|2.51ns|9.72ns|0.496| 0|0|3.15&nbsp;KB|
|#5928|`SendAsync`|net6.0|4.01μs|1.25ns|4.67ns|0.0302| 0|0|2.22&nbsp;KB|
|#5928|`SendAsync`|netcoreapp3.1|5.1μs|2.41ns|9.35ns|0.0384| 0|0|2.76&nbsp;KB|
|#5928|`SendAsync`|net472|7.84μs|2.08ns|7.79ns|0.498| 0|0|3.15&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`EnrichedLog`|net6.0|1.47μs|0.489ns|1.76ns|0.0228| 0|0|1.64&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`EnrichedLog`|netcoreapp3.1|2.16μs|1.33ns|4.96ns|0.0217| 0|0|1.64&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`EnrichedLog`|net472|2.73μs|2.01ns|7.8ns|0.249| 0|0|1.57&nbsp;KB|
|#5928|`EnrichedLog`|net6.0|1.51μs|0.71ns|2.66ns|0.0227| 0|0|1.64&nbsp;KB|
|#5928|`EnrichedLog`|netcoreapp3.1|2.26μs|1.85ns|7.17ns|0.0224| 0|0|1.64&nbsp;KB|
|#5928|`EnrichedLog`|net472|2.64μs|2.52ns|9.77ns|0.249| 0|0|1.57&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`EnrichedLog`|net6.0|116μs|149ns|579ns|0.0584| 0|0|4.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`EnrichedLog`|netcoreapp3.1|119μs|147ns|569ns|0| 0|0|4.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`EnrichedLog`|net472|146μs|173ns|671ns|0.655| 0.218|0|4.46&nbsp;KB|
|#5928|`EnrichedLog`|net6.0|116μs|194ns|752ns|0.0585| 0|0|4.28&nbsp;KB|
|#5928|`EnrichedLog`|netcoreapp3.1|121μs|127ns|476ns|0.0602| 0|0|4.28&nbsp;KB|
|#5928|`EnrichedLog`|net472|149μs|136ns|528ns|0.669| 0.223|0|4.46&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`EnrichedLog`|net6.0|3.15μs|0.813ns|3.15ns|0.0298| 0|0|2.2&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`EnrichedLog`|netcoreapp3.1|4.19μs|1.68ns|6.5ns|0.0294| 0|0|2.2&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`EnrichedLog`|net472|4.92μs|1.19ns|4.46ns|0.32| 0|0|2.02&nbsp;KB|
|#5928|`EnrichedLog`|net6.0|3.06μs|1.19ns|4.44ns|0.0306| 0|0|2.2&nbsp;KB|
|#5928|`EnrichedLog`|netcoreapp3.1|4.26μs|1.9ns|7.37ns|0.0297| 0|0|2.2&nbsp;KB|
|#5928|`EnrichedLog`|net472|4.89μs|1.51ns|5.87ns|0.32| 0|0|2.02&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`SendReceive`|net6.0|1.37μs|1.63ns|6.31ns|0.0158| 0|0|1.14&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`SendReceive`|netcoreapp3.1|1.77μs|1.31ns|5.09ns|0.0151| 0|0|1.14&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`SendReceive`|net472|2.2μs|1.73ns|6.7ns|0.183| 0.0011|0|1.16&nbsp;KB|
|#5928|`SendReceive`|net6.0|1.3μs|0.405ns|1.51ns|0.0163| 0|0|1.14&nbsp;KB|
|#5928|`SendReceive`|netcoreapp3.1|1.76μs|0.975ns|3.78ns|0.015| 0|0|1.14&nbsp;KB|
|#5928|`SendReceive`|net472|2.07μs|0.923ns|3.33ns|0.183| 0.00104|0|1.16&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Faster :tada: Same allocations :heavy_check_mark:



#### Faster :tada: in #5928

|  Benchmark         | base/diff | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.SerilogBenchmark.EnrichedLog&#8209;net6.0 | 1.119 | 3,051.71 | 2,726.82 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`EnrichedLog`|net6.0|3.05μs|4.77ns|18.5ns|0.0226| 0|0|1.6&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`EnrichedLog`|netcoreapp3.1|3.8μs|2.66ns|10.3ns|0.021| 0|0|1.65&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`EnrichedLog`|net472|4.3μs|1.13ns|4.23ns|0.323| 0|0|2.04&nbsp;KB|
|#5928|`EnrichedLog`|net6.0|2.73μs|1.05ns|4.07ns|0.0218| 0|0|1.6&nbsp;KB|
|#5928|`EnrichedLog`|netcoreapp3.1|3.81μs|2.38ns|9.22ns|0.0209| 0|0|1.65&nbsp;KB|
|#5928|`EnrichedLog`|net472|4.4μs|3.89ns|15.1ns|0.322| 0|0|2.04&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #5928

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.SpanBenchmark.StartFinishScope&#8209;netcoreapp3.1 | 1.152 | 686.34 | 790.56 | 
| Benchmarks.Trace.SpanBenchmark.StartFinishSpan&#8209;netcoreapp3.1 | 1.126 | 547.81 | 616.81 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`StartFinishSpan`|net6.0|397ns|0.278ns|1.08ns|0.00803| 0|0|576&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`StartFinishSpan`|netcoreapp3.1|548ns|0.352ns|1.36ns|0.00768| 0|0|576&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`StartFinishSpan`|net472|636ns|0.837ns|3.24ns|0.0918| 0|0|578&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`StartFinishScope`|net6.0|474ns|0.437ns|1.64ns|0.0098| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`StartFinishScope`|netcoreapp3.1|686ns|0.209ns|0.753ns|0.00929| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`StartFinishScope`|net472|823ns|0.83ns|3.21ns|0.104| 0|0|658&nbsp;B|
|#5928|`StartFinishSpan`|net6.0|384ns|0.236ns|0.915ns|0.00807| 0|0|576&nbsp;B|
|#5928|`StartFinishSpan`|netcoreapp3.1|618ns|1.03ns|3.84ns|0.00764| 0|0|576&nbsp;B|
|#5928|`StartFinishSpan`|net472|636ns|0.354ns|1.28ns|0.0916| 0|0|578&nbsp;B|
|#5928|`StartFinishScope`|net6.0|478ns|0.15ns|0.582ns|0.00966| 0|0|696&nbsp;B|
|#5928|`StartFinishScope`|netcoreapp3.1|792ns|1ns|3.61ns|0.00953| 0|0|696&nbsp;B|
|#5928|`StartFinishScope`|net472|853ns|0.482ns|1.87ns|0.104| 0|0|658&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`RunOnMethodBegin`|net6.0|584ns|0.441ns|1.71ns|0.00971| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`RunOnMethodBegin`|netcoreapp3.1|828ns|0.565ns|2.11ns|0.00916| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0b375713b7aa9f1a7d695a24271c0d9e58435fab)|`RunOnMethodBegin`|net472|1.13μs|1.22ns|4.72ns|0.104| 0|0|658&nbsp;B|
|#5928|`RunOnMethodBegin`|net6.0|597ns|0.494ns|1.91ns|0.00983| 0|0|696&nbsp;B|
|#5928|`RunOnMethodBegin`|netcoreapp3.1|882ns|0.606ns|2.27ns|0.0094| 0|0|696&nbsp;B|
|#5928|`RunOnMethodBegin`|net472|1.12μs|1.56ns|6.03ns|0.104| 0|0|658&nbsp;B|
</details>

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -90,13 +90,36 @@ download-single-step-artifacts:
...
 
 package-oci:
   needs: [ download-single-step-artifacts ]
+  artifacts:
+    paths:
+      - "packaging/sources/version"
+      - "packaging/sources/requirements.json"
+      - "packaging/*.tar"
+
+requirements_json_test:
```
Comment (andrewlock) [inline .gitlab-ci.yml:107]: These changes should be ported to one-pipeline, but keeping them for here until the changes are merged to the injector

**Review by kevingosse — APPROVED** [review]

**Review by randomanderson — APPROVED** [review]

**Comment by andrewlock** [comment]
## Throughput/Crank Report :zap:

Throughput results for AspNetCoreSimpleController comparing the following branches/commits:
- [This PR (5928)](https://github.com/DataDog/dd-trace-dotnet/tree/f2d5d45d77ee871a0fc694a685dcc718fa1fc3c1)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/909d3183b047c0d11377a9c47221e53a200bf2bb)
- [benchmarks/2.9.0](https://github.com/DataDog/dd-trace-dotnet/tree/89ae3f4ee7d85b824badda5873166707c9e560b4)

Cases where throughput results for the PR are worse than latest master (5% drop or greater), results are shown in **red**.

Note that these results are based on a _single_ point-in-time result for each branch. For full results, see [one](https://ddstaging.datadoghq.com/dashboard/fnz-afi-c2i/apm-net-crank) of the [many](https://ddstaging.datadoghq.com/dashboard/c5g-i7d-kcy/ciapp-apm-net-throughput-tests), [many](https://ddstaging.datadoghq.com/dashboard/uxh-m8j-qhi/ciapp-apm-net-throughput-tests-kevin) dashboards!

```mermaid
gantt
    title Throughput Linux x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5928) (10.887M)   : 0, 10887162
    master (10.964M)   : 0, 10963679
    benchmarks/2.9.0 (11.081M)   : 0, 11080577

    section Automatic
    This PR (5928) (7.158M)   : 0, 7158417
    master (7.342M)   : 0, 7341975
    benchmarks/2.9.0 (7.732M)   : 0, 7732233

    section Trace stats
    master (7.624M)   : 0, 7624069

    section Manual
    master (11.004M)   : 0, 11004392

    section Manual + Automatic
    This PR (5928) (6.583M)   : 0, 6583437
    master (6.893M)   : 0, 6893295

    section DD_TRACE_ENABLED=0
    master (10.114M)   : 0, 10113707

```
```mermaid
gantt
    title Throughput Linux arm64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5928) (9.336M)   : 0, 9336274
    master (9.408M)   : 0, 9407832
    benchmarks/2.9.0 (9.798M)   : 0, 9798067

    section Automatic
    This PR (5928) (6.616M)   : 0, 6616191
    master (6.464M)   : 0, 6463516

    section Trace stats
    master (6.767M)   : 0, 6767065

    section Manual
    master (9.424M)   : 0, 9424497

    section Manual + Automatic
    This PR (5928) (6.177M)   : 0, 6176756
    master (6.104M)   : 0, 6104219

    section DD_TRACE_ENABLED=0
    master (8.750M)   : 0, 8750160

```
```mermaid
gantt
    title Throughput Windows x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5928) (10.148M)   : 0, 10148424
    master (10.312M)   : 0, 10312327
    benchmarks/2.9.0 (10.067M)   : 0, 10067315

    section Automatic
    This PR (5928) (6.912M)   : 0, 6912111
    master (6.814M)   : 0, 6814056
    benchmarks/2.9.0 (7.552M)   : 0, 7552193

    section Trace stats
    master (7.412M)   : 0, 7412240

    section Manual
    master (10.113M)   : 0, 10113182

    section Manual + Automatic
    This PR (5928) (6.242M)   : 0, 6242347
    master (6.441M)   : 0, 6440538

    section DD_TRACE_ENABLED=0
    master (9.598M)   : 0, 9597874

```

File: tracer/build/artifacts/requirements.json [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    "glibc": [{
+      "arch": "x64",
+      "supported": true,
+      "description": "From centOS 7",
+      "min": "2.17"
+    },
+    {
+      "arch": "arm64",
+      "supported": true,
+      "description": "From debian buster (10)",
```
Comment (andrewlock) [inline tracer/build/artifacts/requirements.json:14]:
```suggestion
      "description": "From debian buster (10), which is 2.28, but we verify we don't exceed 2.23",
```

**Pull request merged**